pytorch_lightning/core/decorators.py

```diff
@@ -71,12 +71,11 @@ def auto_transfer_args(self, *args, **kwargs):
 
 def parameter_validation(fn: Callable) -> Callable:
```

Comment:
I think now that you changed the decorator target to self.model, this decorator may no longer fit very well into core/decorators because it is basically now specific to the plugin having the attribute self.model. 
What do you think about moving it? 

Just for consideration

Reply:
+1 to @awaelchli 's suggestion

Reply:
Good catch. Will do a follow-up PR for this.